### PR TITLE
Replace all the tokens in the jobcall back headers and body

### LINKED ISSUE #559: Job call back params in the body are only replaced for the first instance and not all instances
Hi,
I have the following configuration in my job file
job.notification.started.1.url=${web.hookl}
job.notification.started.1.method=POST
job.notification.started.1.body='Execution ?{executionId} of flow ?{flow} in project ?{project} has ?{status} on server ?{server}. Execution ?{executionId} of flow ?{flow} in project ?{project} has ?{status} on server ?{server}.'
job.notification.started.1.headers=Content-type:application/json

When I get the email, I see the following in the body
'Execution 64 of flow analytics in project Test has queued on server WeveJobs01. Execution ?{executionId} of flow ?{flow} in project ?{project} has ?{status} on server ?{server}.'

Which means only the frist instance of the replacables ?{server}, ?{flow} etc are being replaces. I would expect all instances to be replaced

### PR DESCRIPTION
**Pull request by vikramkone**
Currently we are only replacing the first instance of the tokens in the header and bbody for job call back. This doesn't work in cases where we want to craft a body or header which needs the token to be replaced at all places.

Fixes #559

### CODE REVIEW

**Comment by vikramkone** [comment]
Hi @davidzchen @hluu ..this is my first time contributiing back to azkaban. Can you let me know wy the travis build is failing for oracle jdk and jdk8 but passing for jdk 7? What do I need to change to make the build work across all JDK versions?

**Comment by davidzchen** [comment]
That's strange. It doesn't seem to be anything related to your patch; the error seems to be caused by some connection failure when trying to fetch artifacts from Maven.

I am rerunning the CI to see if it goes away.

File: azkaban-execserver/src/main/java/azkaban/execapp/event/JobCallbackUtil.java [hunk 1]
```diff
@@ -181,6 +181,13 @@ public static boolean isThereJobCallbackProperty(Props props,
         String headersKeyPerSequence =
             headersKey.replace(SEQUENCE_TOKEN, sequenceStr);
         String headersValue = props.get(headersKeyPerSequence);
+		
+	// replace all the tokens in the header
+	if(headersValue != null)
```
Comment (davidzchen) [inline azkaban-execserver/src/main/java/azkaban/execapp/event/JobCallbackUtil.java]: Nit: Make indentation and brace style consistent.

**Comment by davidzchen** [comment]
LGTM

**Comment by hluu** [comment]
Can you add some tests?

**Comment by vikramkone** [comment]
Added unit test and fixed indentation

**Comment by hluu** [comment]
LGTM

**Comment by vikramkone** [comment]
Thanks for the review. Do I need to do anything else from my side, to merge the request into the master?

File: azkaban-execserver/src/main/java/azkaban/execapp/event/JobCallbackUtil.java [hunk 2]
```diff
@@ -181,6 +181,12 @@ public static boolean isThereJobCallbackProperty(Props props,
         String headersKeyPerSequence =
             headersKey.replace(SEQUENCE_TOKEN, sequenceStr);
         String headersValue = props.get(headersKeyPerSequence);
+		
+		// replace all the tokens in the header
```
Comment (davidzchen) [inline azkaban-execserver/src/main/java/azkaban/execapp/event/JobCallbackUtil.java]: Nit: Please fix indentation

File: azkaban-execserver/src/test/java/azkaban/execapp/event/JobCallbackUtilTest.java [hunk 3]
```diff
@@ -211,6 +211,31 @@ public void allTokensTest() {
   }
 
   @Test
+  public void multipleTokensTest() {
```
Comment (davidzchen) [inline azkaban-execserver/src/test/java/azkaban/execapp/event/JobCallbackUtilTest.java]: Nit: indentation

**Comment by davidzchen** [comment]
Thanks, @vikramkone! I only have two minor style comments. Other than that, can you squash your commits into a single commit using `git rebase -i`? Then, I'll go ahead and merge your patch to master.

**Comment by davidzchen** [comment]
Hi @vikramkone, I noticed that your pull request is based off of your `master` branch. As a result, I don't think you would be able to do a straightforward rebase this way. Generally, it is better to make changes for a bug on a feature branch and keep your `master` identical to the upstream `master`. This way, you would be able to work on more than one change simultaneously, and it would make it easier to manage merge conflicts. Plus, Git was designed with this model in mind.

I have a write-up about how to move your changes to a feature branch and make your `master` branch in sync with the upstream master at: https://github.com/azkaban/azkaban/pull/517#issuecomment-146736037.

Feel free to let me know if you have any questions.

**Comment by vikramkone** [comment]
@davidzchen  Made the changes as mentioned in your other thread...help review and merge

**Comment by juhoautio** [comment]
5/5, would use :+1:

**Comment by suvodeep-pyne** [comment]
Cleaning up unattended / old PRs

**Pull request closed without merging**